app/src/hooks/useParaspellApi.tsx

```diff
@@ -155,22 +157,35 @@ const useParaspellApi = () => {
     }
   }
 
-  const validate = async (params: TransferParams) => {
+  interface DryRunResult {
+    dryRunSupported: boolean
+    success: boolean
+    error?: string
+    fee?: bigint
+  }
```

Comment:
This would be a lot nicer with a union type  👀 

```suggestion
  type DryRunResult = { type: 'Supported', result: boolean, fees: bigint } | { type: 'Unsupported', error?: string }
```

Reply:
Another option could be to 'combine' PS `TDryRunResult` type instead of creating a new similar one? It might introduce more complexity to create the response though..
 
`type DryRunResult = { type: 'Supported' | 'Unsupported'} & TDryRunResult`